### PR TITLE
refactor: Use stage 3 decorators

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

Use modern decorator implementation with stricter types

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
Seems like coverage tooling doesn't handle decorators well, feel free to suggest the solution better than `istanbul ignore`

File: lib/modules/datasource/aws-machine-image/index.ts [hunk 1]
```diff
@@ -11,8 +11,10 @@ import { Datasource } from '../datasource';
 import type { GetReleasesConfig, ReleaseResult } from '../types';
 import type { AwsClientConfig, ParsedConfig } from './types';
 
+const id = 'aws-machine-image';
```
Comment (viceice) [inline lib/modules/datasource/aws-machine-image/index.ts:14]: why this change? if it's required then please do in separate PR, otherwise this PR is hard to review because of big changes

File: lib/modules/datasource/aws-machine-image/index.ts [hunk 2]
```diff
@@ -80,11 +82,13 @@ export class AwsMachineImageDataSource extends Datasource {
     return [filters, config];
   }
 
-  @cache({
-    namespace: `datasource-${AwsMachineImageDataSource.id}`,
-    key: (serializedAmiFilter: string) =>
-      `getSortedAwsMachineImages:${serializedAmiFilter}`,
-  })
+  @cache(
+    /* istanbul ignore next */ {
```
Comment (viceice) [inline lib/modules/datasource/aws-machine-image/index.ts:86]: why do we need to ignore this now? 😕

File: lib/modules/datasource/aws-machine-image/index.ts (hunk 1 above)
Comment (zharinov) [reply lib/modules/datasource/aws-machine-image/index.ts:14]: Stage 3 decorators seem to be instantiated before the class itself, so static constants are not available

Comment (viceice) [reply lib/modules/datasource/aws-machine-image/index.ts:14]: ok, please split that refactor to a separate PR, so we've less changes.
